src/services/usercentrics-service.ts

```diff
@@ -27,36 +30,59 @@ const subscribe = (x: (x: ServiceState) => void): (() => void) => {
     }
 }
 
-const setIsOpen = (isOpen: boolean) => {
-    state = { ...state, isOpen }
+const setIsOpen = (isOpen: boolean, hasInteracted: boolean) => {
+    state = { ...state, isOpen, hasInteracted }
     update()
 }
-const setIsInitialized = (initialized: boolean) => {
-    state = { ...state, initialized }
+
+const setIsInitialized = (isInitialized: boolean) => {
+    state = { ...state, isInitialized }
     update()
 }
 
 const onEvent = (event: Event) => {
     if (isUCUICMPEvent(event)) {
         switch (event.detail.type) {
-            case UCUICMPEventType.CMP_SHOWN:
-                setIsOpen(true)
+            case UCUICMPEventType.CMP_SHOWN: {
+                setIsOpen(true, hasUserInteracted())
```

Comment:
This might not be necessary, since `initializeState()` runs before this, and simply opening the modal doesn't imply interaction (if user simply closes the page without interacting with it).